code/go/0chain.net/smartcontract/storagesc/allocation_test.go

```diff
@@ -2438,8 +2434,372 @@ func TestUpdateAllocationRequest(t *testing.T) {
+
+		// Define the number of blobbers that will increase or decrease their price
+		increasePriceCount := 5 // Number of blobbers to increase the price for
+		decreasePriceCount := 2 // Number of blobbers to decrease the price for
+
+		var totoalCoinsRequiredForUpgrade currency.Coin
+		var totalWritePriceBeforeAlloc currency.Coin
+		var sizePerBlobber = currency.Coin(sizeInGB(beforeAlloc.Size / int64(beforeAlloc.DataShards)))
+
+		// Iterate over all blobbers and adjust their write price
+		oldTerms := make([]Terms, 0)
+		for i, blobber := range blobbers {
+
+			blobberNode, err := ssc.getBlobber(blobber.id, balances)
+			require.NoError(t, err)
+
+			oldTerms = append(oldTerms, blobberNode.mustBase().Terms)
+
+			// Add current blobber's write price to total
+			totalWritePriceBeforeAlloc += (sizePerBlobber * blobberNode.mustBase().Terms.WritePrice)
+
+			// Adjust price: first `increasePriceCount` blobbers increase price by 2x, next `decreasePriceCount` blobbers decrease price by 0.5x
+			if i < increasePriceCount {
+				blobberNode.mustUpdateBase(func(b *storageNodeBase) error {
+					b.Terms.WritePrice = b.Terms.WritePrice * 2
+					return nil
+				})
+			} else if i >= increasePriceCount && i < increasePriceCount+decreasePriceCount {
+				blobberNode.mustUpdateBase(func(b *storageNodeBase) error {
+					b.Terms.WritePrice = b.Terms.WritePrice / 2
+					return nil
+				})
+			}
+			_, err = updateBlobber(t, blobberNode, 0, tp, ssc, balances)
+			require.NoError(t, err)
+			// Add the new blobber price to the total lock amount
+			totoalCoinsRequiredForUpgrade += (sizePerBlobber * blobberNode.mustBase().Terms.WritePrice)
+		}
+
+		// Prepare for allocation upgrade
+		var uar updateAllocationRequest
+		uar.ID = allocID
+		uar.Size = 10 * GB
+		// 1296000 seconds = 360 hours
+		tp += int64(360 * time.Hour / 1e9)
+
+		// First attempt to upgrade allocation without providing enough lock amount (should fail)
+		// resp, err := uar.callUpdateAllocReq(t, client.id, 0, tp, ssc, balances)
+		// require.Error(t, err)
+
+		//todo:
+		// get the time duration from the start of transaction
+		// and update the challneg pool amount
+
+		cp, err := ssc.getChallengePool(allocID, balances)
+		require.NoError(t, err)
+
+		//cp.Balance = cp.Balance / 2
+
+		// Calculate the total expected lock amount and upgrade lock amount based on size multiplier
+		sizeMultiplier := float64(uar.Size+beforeAlloc.Size) / float64(beforeAlloc.Size)
+		totalExpectedLockAmount := currency.Coin(float64(totoalCoinsRequiredForUpgrade) * sizeMultiplier)
+		totalUpgradeLockAmount := currency.Coin(float64(totalExpectedLockAmount) - (float64(totalWritePriceBeforeAlloc) - float64(cp.Balance/2)))
+		totalExpectedSize := uar.Size + beforeAlloc.Size
 
+		// Now provide the correct lock amount for the upgrade (should succeed)
+		resp, err := uar.callUpdateAllocReq(t, client.id, totalUpgradeLockAmount, tp, ssc, balances)
```

Comment:
Add challenge before update allocation and verify challenge pool changes in during update allocation